ngraph_bridge/ngraph_builder.cc

```diff
@@ -913,9 +913,37 @@ static Status TranslateBatchMatMulOp(
   return Status::OK();
 }
 
-static Status TranslateBiasAddOp(const Node* op,
-                                 const std::vector<const Tensor*>&,
-                                 Builder::OpMap& ng_op_map) {
+static Status TranslateBatchMatMulV2Op(
+    const Node* op, const std::vector<const Tensor*>& static_input_map,
+    Builder::OpMap& ng_op_map) {
+  shared_ptr<ng::Node> ng_lhs, ng_rhs;
+  TF_RETURN_IF_ERROR(GetInputNodes(ng_op_map, op, &ng_lhs, &ng_rhs));
+
+  auto ng_lhs_shape = ng_lhs->get_shape();
+  auto ng_rhs_shape = ng_rhs->get_shape();
+  size_t n_dims = ng_lhs_shape.size();
+
+  for (size_t i = 0; i < n_dims - 2; ++i) {
+    if (!((ng_lhs_shape[i] == ng_rhs_shape[i]) || (ng_lhs_shape[i] == 1) ||
+          (ng_rhs_shape[i] == 1))) {
+      return errors::InvalidArgument(
+          "ng_lhs_shape and ng_rhs_shape must be broadcastable for "
+          "BatchMatMulV2 "
+          "for each dimension",
+          i);
+    } else if (ng_lhs_shape[i] != ng_rhs_shape[i]) {
+      return errors::Unimplemented(
+          "ng_lhs_shape and ng_rhs_shape must be the same for BatchMatMulV2 "
+          "for each dimension",
+          i);
```

Comment:
I think the two `if` statements are necessary because, in the first one, we are checking if the shapes are not the same and not broadcastable than `InvalidArgument` is returned. But in the second check, the only availability of broadcasting is checked, and if it is broadcastable `Unimplemented` is returned. I think this is the essential difference. But I also changed the error message to be more informative and more diverse between these cases.

Reply:
Ah, I see your point.
check 1 is for broadcastability (which TF allows)
but since we don't support it, we return Unimplemented.

The 2 ifs are fine then. Maybe we can change the error message a bit in the 2 cases to make that explicit